client/packages/invoices/src/Returns/DetailView/Footer/StatusChangeButton.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useMemo, useState, useEffect } from 'react';
+import {
+  ArrowRightIcon,
+  useTranslation,
+  useNotification,
+  InvoiceNodeStatus,
+  SplitButton,
+  SplitButtonOption,
+  useConfirmationModal,
+} from '@openmsupply-client/common';
+import { getNextOutboundStatus, getStatusTranslation } from '../../../utils';
+import { useReturns } from '../../api';
+
+const getStatusOptions = (
+  currentStatus: InvoiceNodeStatus,
+  getButtonLabel: (status: InvoiceNodeStatus) => string
+): SplitButtonOption<InvoiceNodeStatus>[] => {
+  const options: [
+    SplitButtonOption<InvoiceNodeStatus>,
+    SplitButtonOption<InvoiceNodeStatus>,
+    SplitButtonOption<InvoiceNodeStatus>,
+    SplitButtonOption<InvoiceNodeStatus>,
+    SplitButtonOption<InvoiceNodeStatus>,
+    SplitButtonOption<InvoiceNodeStatus>,
+  ] = [
+    {
+      value: InvoiceNodeStatus.New,
+      label: getButtonLabel(InvoiceNodeStatus.New),
+      isDisabled: true,
+    },
+    {
+      value: InvoiceNodeStatus.Allocated,
+      label: getButtonLabel(InvoiceNodeStatus.Allocated),
+      isDisabled: true,
+    },
+    {
+      value: InvoiceNodeStatus.Picked,
+      label: getButtonLabel(InvoiceNodeStatus.Picked),
+      isDisabled: true,
+    },
+    {
+      value: InvoiceNodeStatus.Shipped,
+      label: getButtonLabel(InvoiceNodeStatus.Shipped),
+      isDisabled: true,
+    },
+    {
+      value: InvoiceNodeStatus.Delivered,
+      label: getButtonLabel(InvoiceNodeStatus.Delivered),
+      isDisabled: true,
+    },
+    {
+      value: InvoiceNodeStatus.Verified,
+      label: getButtonLabel(InvoiceNodeStatus.Verified),
+      isDisabled: true,
+    },
+  ];
+
+  if (currentStatus === InvoiceNodeStatus.New) {
+    options[1].isDisabled = false;
+    options[2].isDisabled = false;
+    options[3].isDisabled = false;
+  }
+
+  if (currentStatus === InvoiceNodeStatus.Allocated) {
+    options[2].isDisabled = false;
+    options[3].isDisabled = false;
+  }
+
+  if (currentStatus === InvoiceNodeStatus.Picked) {
+    options[3].isDisabled = false;
+  }
+
+  return options;
+};
+
+const getNextStatusOption = (
+  status: InvoiceNodeStatus,
+  options: SplitButtonOption<InvoiceNodeStatus>[]
+): SplitButtonOption<InvoiceNodeStatus> | null => {
+  if (!status) return options[0] ?? null;
+
+  const nextStatus = getNextOutboundStatus(status);
+  const nextStatusOption = options.find(o => o.value === nextStatus);
+  return nextStatusOption || null;
+};
```

Comment:
Again, can probably pull out into shared functions?

Reply:
Agree, this would be great to create a refactor issue to pull out 👍

Reply:
https://github.com/msupply-foundation/open-msupply/issues/3203 :)